webtransport/README.md

```diff
@@ -0,0 +1,52 @@
+# libp2p WebTransport
+
+| Lifecycle Stage | Maturity                 | Status | Latest Revision |
+|-----------------|--------------------------|--------|-----------------|
+| 1A              | DRAFT                    | Active | r0, 2022-04-03  |
+
+Authors: [@marten-seemann]
+
+Interest Group: TODO
+
+See the [lifecycle document](../00-framework-01-spec-lifecycle.md) for context about maturity level
+and spec status.
+
+## Introduction
+
+WebTransport is a way for browsers to establish a stream-multiplexed and bidirectional connection to servers using QUIC.
+
+The WebTransport protocol is currently under development at the IETF. Chrome has implemented and shipped support for [draft-02](https://datatracker.ietf.org/doc/draft-ietf-webtrans-http3/).
+
+The most exciting feature for libp2p (other than the improved performance and hole punching success rates that QUIC gives us) is that the W3C added a browser API allowing browsers to establish connections to nodes with self-signed certificates, provided they know the hash of the certificate in advance: [`serverCertificateHashes`](https://www.w3.org/TR/webtransport/#dom-webtransportoptions-servercertificatehashes). This API is already [implemented in Chrome](https://chromestatus.com/feature/5690646332440576). Firefox is working on a WebTransport implementation and is [likely to implement](https://github.com/mozilla/standards-positions/issues/167#issuecomment-1015951396) `serverCertificateHashes` as well.
```

Comment:
> and hole punching success rates

Will we ever be able to leverage WebTransport for hole punching? Would the browsers allow us to reuse the same UDP port across connections?

Reply:
In principle it's possible to demultiplex QUIC connections by their Connection ID. In fact, that's what go-libp2p does. Currently, browsers use a new UDP socket per connection, but that might change in the future (maybe? hopefully?).

We should probably remove this sentence though, since it's not applicable today yet.

Reply:
Done.